src/others/components/map/Map.tsx

```diff
@@ -78,11 +86,14 @@ export const Map = ({ sourceWithLayer }: MapProps) => {
         style={{ borderRadius: "24px" }}
         interactiveLayerIds={["ukr_water_needs-point"]}
         cursor={cursor}
+        onLoad={() => {setMapLoaded(true);}}
         onMouseEnter={handleMouseEnter}
         onMouseLeave={handleMouseLeave}
       >
         {sourceWithLayer}
 
+        {showRegions && <RegionsSource aidRequests={aidRequests} mapRef={mapRef} mapLoaded={mapLoaded} />}
```

Comment:
This is kinda problematic in the scope of having this component reusable for incidents map (and that was the purpose of using {sourceWithLayer} in props. What I was thinking of is in case of multiple layers that we would like to have, we could pass a list of children layers and declare them here in the loop.

Of course, if we are giving up the idea of having this component reusable for the incidents map we should add a circle layer in a similar way to the regions layer.

One of those two options would work but I'm definitely going for not mixing two approaches here. I also apologize for raising it right now, should've done it before in case of tooltips...

Reply:
LGTM :)

Reply:
As for incidents, at some point, we will just fork from this repo and continue on our own, since UGT cannot be directly involved with the incidents project anymore and we have to branch away.